backend/sportsmatch/src/main/java/com/sportsmatch/services/UserServiceImp.java

```diff
@@ -76,11 +78,23 @@ private void parseUserDateOfBirth(UserInfoDTO userInfoDTO, User user) {
 
   private void linkUserWithSport(UserInfoDTO userInfoDTO, User user) {
     for (SportDTO sportDTO : userInfoDTO.getSports()) {
-      if (sportRepository.findSportByName(sportDTO.getName()).isEmpty()) {
-        throw new ResponseStatusException(HttpStatus.NOT_FOUND);
+      Optional<Sport> optionalSport = sportRepository.findSportByName(sportDTO.getName());
+
+      if (optionalSport.isPresent()) {
+        Sport sport = optionalSport.get();
+
+        boolean isAssociated =
+            user.getSportUsers().stream()
+                .anyMatch(sportUser -> Objects.equals(sportUser.getSport().getId(), sport.getId()));
+
+        if (!isAssociated) {
+          SportUser sportUser = new SportUser(user, sport);
+          user.getSportUsers().add(sportUser);
+        }
+      } else {
+        throw new ResponseStatusException(
```

Comment:
Is it neccessary to throw an exception here? Because if one of the sports is not found on the backend, it will fail to update user info. And user won't get feedback that sport they selected does not exist.
On the other hand if available sports will be retieved from the database, it should not happen
![image](https://github.com/MatejFrnka/sportsmatch/assets/131284389/285e415f-d555-462f-af50-bbaab3f834e6)